### PR TITLE
[3.0] Fixes custom normalizer usage

### LINKED ISSUE #144: [3.0] Custom Normalizer is not called
Given a customized `Normalizer` (that relies on other `Normalizers`) defined as a service & registered in the Symfony app.
I would  expect the search-bundle to use it and but that's not the case.

```yml
algolia_search:
    indices:
        - name: cv
          class: AppBundle\Entity\Cv
          normalizers:
            - AppBundle\Serializer\Normalizer\CvNormalizer
```

```php
<?php

namespace AppBundle\Serializer\Normalizer;

use AppBundle\Entity\Cv;
use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
use Symfony\Component\Serializer\SerializerAwareInterface;
use Symfony\Component\Serializer\SerializerAwareTrait;

class CvNormalizer implements NormalizerInterface, SerializerAwareInterface
{
    use SerializerAwareTrait;

    public function normalize($object, $format = null, array $context = [])
    {
        /* @var Cv $object */

        if ($format == 'searchableArray') {
            if ($object->getExpiresAt() < new \DateTime("18 months ago")) {
                return [];
            }
        }

        return [
            'id'        => $object->getId(),
            'name'      => $object->getPseudo(),
            'pseudo'    => $object->getPseudo(),
            'slug'      => $object->getSlug(),
            'createdAt' => $this->serializer->normalize($object->getCreatedAt(), $format, $context),
            'updatedAt' => $this->serializer->normalize($object->getUpdatedAt(), $format, $context),
            'user'      => $this->serializer->normalize($object->getUser(), $format, $context),
            'profiles'  => $this->serializer->normalize($object->getProfiles(), $format, $context),
            'game'      => $this->serializer->normalize($object->getGame(), $format, $context),
            'platform'  => $this->serializer->normalize($object->getPlatform(), $format, $context),
        ];
    }

    public function supportsNormalization($data, $format = null)
    {
        return $data instanceof Cv;
    }
}
```

```xml
<service id="cv_normalizer" class="AppBundle\Serializer\Normalizer\CvNormalizer" public="false">
    <tag name="serializer.normalizer" />
</service>
```

## Result 

Default Algolia normalizer is used instead of mine 

hosted_cloud_search_as_a_service___algolia

### PR DESCRIPTION
**Pull request by tristanbes**
TLDR: You shouldn't merge this PR, merge #153 instead ✌️

---

This PR solves 2 problems:

1. Custom normalizer not being called (#144)
2. Normalizer passed as class name instead of object instance (#148). this one is little more complicated than just instanciating the object via the `new` keyword because you can define `Normalizer` as a service and in that case the `Normalizer` would break. So I had to inject the container instead and loop on service id.

Note that it has some (**MAJOR**) drawbacks:
- You have to set the `Normalizer` attribute `public=true` when defining your custom `Normalizer` service.
- You have to overwrite built-in Symfony Normalizer services to expose them as public, otherwise you cannot use them.
- You have to EXPLICITLY tell all the `Normalizers` dependencies. If you object has for example 10 relations to other objects, you have to add those 10 `Normalizers` in your indice configuration

Fix #144 and #148 

Disclaimer: I didn't test if I broke indexing entities via implementing the `normalize` method on the entity itself.

Also, after thinking through this on my way home, I've decided that **#153 is a much more elegent solution**.

### CODE REVIEW

**Comment by julienbourdeau** [comment]
Closing because #153 seems like a much better solution. Thanks for your contributions!
